Mathlib/FieldTheory/Minpoly/Field.lean

```diff
@@ -126,19 +126,29 @@ theorem eq_of_irreducible [Nontrivial B] {p : A[X]} (hp1 : Irreducible p)
   · rwa [Polynomial.Monic, leadingCoeff_mul, leadingCoeff_C, mul_inv_cancel]
 #align minpoly.eq_of_irreducible minpoly.eq_of_irreducible
 
+/-- If `y` is the image of `x` under an injective algebra homomorphism, their minimal polynomials
+coincide.
+
+We take `h : y = f x` as an argument because `rw [h]` typically fails
+since `IsIntegral K y` depends on y.
```

Comment:
I find this comment weird: where does `IsIntegral K y` even appear? If it does, why can't you rewrite it by `h` too?
From [search results](https://github.com/search?q=repo%3Aleanprover-community%2Fmathlib4%20eq_of_algebraMap_eq&type=code) it seems `h = rfl` most of the time.

Reply:
I also find it weird, but it's just a generalization of a previous lemma with this comment, so I'm happy to merge as is and let someone fix it in a follow-up.

bors d+